### PR TITLE
add option to configure env vars in eval/test config

### LINKED ISSUE #105: Request to Provide Environment Variables for Eval Template/Configuration
# What would you like to be added
When using environment variables to make our rego files more dynamic, it would be nice to be able to provide environment variables on the configuration template of the opa-idea-plugin. 

We have the following in our rego file:
```
my_env = output {
  output = opa.runtime().env
}
```
and in order to provide the environment variable in the eval execution, we have to drop to a terminal and execute the plugins output prefixed by the environment variable and value.
```
OUR_ENV_VAR=value opa eval -f pretty -b /bundle/dir/opa-rules --input /bundle/dir/opa-rules/path-to/input-template.json data.my_package_.deny
```

Note other plugins have this configuration available (when exported):
```
    <envs>
      <env name="OUR_ENV_VAR" value="value" />
    </envs>
``` 

Using: 0.6.0.0-211 in IntelliJ 2021.1.1
# Why is this needed
Currently there no way of setting environment variables to be passed to the command execution environment through the plugin.

# Implementation details

### PR DESCRIPTION
**Pull request by asadali**
# Description
- add option to configure env vars in eval/test config
- bump version for release to 0.7

Fixes #105

### CODE REVIEW

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 package org.openpolicyagent.ideaplugin.ide.runconfig.test
 
+import com.google.common.collect.ImmutableMap
```
Comment (anderseknert) [inline src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt:8]: Why is this needed here?

File: src/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/OpaEvalRunProfileState.kt [hunk 2]
```diff
@@ -35,6 +35,7 @@ class OpaEvalRunProfileState(
         args.add(runConfiguration.query)
 
         val cmd = GeneralCommandLine(OpaBaseTool.opaBinary)
+            .withEnvironment(runConfiguration.env.envs)
```
Comment (anderseknert) [inline src/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/OpaEvalRunProfileState.kt:38]: Surely some IntelliJ thing, but where are the `envs` defined here? Also, a method called `withEnvironment` sounds like it would take a single environment, but `envs` sounds like it would be a collection?

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt [hunk 3]
```diff
@@ -136,6 +143,7 @@ class OpaTestRunConfigurationTest : OpaTestRunConfigurationBase() {
 
     private fun validBundleDir(): Path = Paths.get("${myFixture.tempDirPath}/${bundleDirName}")
     private fun validAdditionalArgs() = "-f pretty -v -t 12s"
+    private fun validEnvs() = EnvironmentVariablesData.DEFAULT.with(ImmutableMap.of("CUSTOM_ENV", "custom-value"))
```
Comment (vgramer) [inline src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt:146]: kotlin has an immutable map built-in [mapOf()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-of.html) ;)

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt (hunk 1 above)
Comment (asadali) [reply src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt]: switched to mapOf.
this was my lack of kotlin showing through :)

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt (hunk 3 above)
Comment (asadali) [reply src/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationTest.kt]: awesome. switched to using that instead.

File: src/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/OpaEvalRunProfileState.kt (hunk 2 above)
Comment (asadali) [reply src/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/OpaEvalRunProfileState.kt:38]: > where are the envs defined here? 

envs are retrieved from `EnvironmentVariablesData.getEnvs` [link](https://github.com/JetBrains/intellij-community/blob/master/platform/execution-impl/src/com/intellij/execution/configuration/EnvironmentVariablesData.java#L43-L49)

> Also, a method called withEnvironment sounds like it would take a single environment, but envs sounds like it would be a collection?

TL;DR: :shrug: 
`withEnvironment` accepts a map<String, String>. runConfiguration.env is an abstraction of env vars called `EnvironmentVariablesData` which exposes a map in the form of `envs` variable.

**Comment by asadali** [comment]
ready for a re-review..

**Review by anderseknert — APPROVED** [review]
